### PR TITLE
crashdump dataset not mounted when generating crash dumps

### PR DESCRIPTION
**Pull request by grwilson**
I've opened up a [PR](https://github.com/delphix/appliance-build/pull/293) for appliance-build repo too.

### CODE REVIEW

**Comment by grwilson** [comment]
[ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-stage1/job/master/job/pre-push/1877/)

File: var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml [hunk 1]
```diff
@@ -96,6 +95,20 @@
...
+# that service will grab the 'spa_namespace_lock' and hold it for a long time.
+# This can result in a systemd timing out and a failure of the local mount
+# service. In addition, we need to run before the 'kdump-tools' service
+# since that service needs to write into '/var/crash'.
+#
+- mount:
+    path: /var/crash
+    src: rpool/crashdump
+    fstype: zfs
+    opts: x-systemd.before=zfs-import-cache.service,x-systemd.before=kdump-tools.service
```
Comment (prakashsurya) [inline var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml]: tangential question.. do we need these mount options for our other datasets managed by `/etc/fstab`?

Comment (grwilson) [reply var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml]: We need one of them and I'm filing a PR for it now.

Comment (prakashsurya) [reply var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml]: Nice, thank you.

Comment (grwilson) [reply var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml]: I just realized that I did not include `defaults` in the options list so I'll update this PR

**Comment by grwilson** [comment]
Updated [ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/1129/)

**Comment by grwilson** [comment]
Updated [ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/1164/)

**Comment by grwilson** [comment]
Updated [ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/1164/)

**Comment by grwilson** [comment]
I spent some time trying to understand why we had crashdump information both in `delphix-platform` and `appliance-build`. This was required because when `delphix-platform` is installed we are still building the image in a local directory (binary directory) and not directly on a ZFS pool so we can't run `zfs` commands yet.  This means that we do most of the setup in `appliance-build` after we've created rpool and rely on `delphix-platform` to reconfigure the crashdump service upon first boot or upgrade.

**Pull request merged**